Watchman.DomainModel/Help/Queries/Handlers/GetHelpInformationQueryHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Watchman.Cqrs;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Help.Queries.Handlers
+{
+    public class GetHelpInformationQueryHandler : IQueryHandler<GetHelpInformationQuery, GetHelpInformationQueryResult>
+    {
+        private const string _helpFileName = "helpInformation.json";
+
+        public GetHelpInformationQueryResult Handle(GetHelpInformationQuery query)
+        {
+            var serverHelpInformations = GetHelpInformationFromDb(query.Session, query.ServerId).ToList();
+            var defaultHelpInformations = GetDefaultHelpInformation(query.Session);
+
+            var helpInformations = new List<DefaultHelpInformation>();
+
+            foreach (var defaultHelp in defaultHelpInformations)
+            {
+                // todo: check what is better: SequenceEqual or Equals
+                var serverHelp = serverHelpInformations.FirstOrDefault(x => x.MethodNames.SequenceEqual(defaultHelp.MethodNames));
+                helpInformations.Add(serverHelp ?? defaultHelp);
+            }
+
+            return new GetHelpInformationQueryResult(helpInformations);
+        }
+
+        private IEnumerable<ServerHelpInformation> GetHelpInformationFromDb(ISession session, ulong serverId)
+        {
+            return session.Get<ServerHelpInformation>()
+                .Where(x => x.ServerId == serverId);
+        }
```

Comment:
Ta sama sytuacja co niżej
bez sensu jest tworzenie osobnej metody, to zapytania które jasno pokazuje co robi, a w dodatku jest bardzo krótkie

Reply:
Czyli wrzucić do metody wywołującej?
**Do zrobienia.**

Reply:
Używaj po prostu session.Get<ServerHelpInformation>().Where(x => x.ServerId == serverId);

ten kod nie wymaga osobnej metody bo jest prosty i czytelny